samples/istio/gitops/templates/experiment.yaml

```diff
@@ -13,36 +13,28 @@ spec:
     actions:
       # when the experiment completes, promote the winning version in the Env repo
       finish:
-      - task: common/exec
+      - task: notification/http
         with:
-          cmd: /bin/bash
-          args: [
-                 "-c",
-                 "apt-get install -y git jq curl;\
-                  REPO=github.com/MY_ORG/iter8;\
-                  BRANCH=master;\
-                  USER=MY_ORG;\
-                  TOKEN=`kubectl -n {{ .namespace }} get secret iter8-token -o json | jq -r .data.token | base64 -d`;\
-                  git config --global user.email 'iter8@iter8.tools';\
-                  git config --global user.name 'Iter8';\
-                  git clone https://${USER}:${TOKEN}@${REPO} --branch=${BRANCH};\
-                  cd iter8/samples/istio/gitops;\
-                  TMP=`mktemp`;\
-                  sed 's/candidate/stable/g' {{ .filepath }} > $TMP;\
-                  mv $TMP productpage.yaml;\
-                  rm -f productpage-candidate.yaml;\
-                  rm -f fortio.yaml;\
-                  rm -f experiment.yaml;\
-                  if [ `git status -s | wc -l` != 0 ];\
-                  then \
-                    git checkout -b iter8_exp;\
-                    git commit -a -m 'update baseline';\
-                    git push -f origin iter8_exp;\
-                    curl -u${USER}:${TOKEN} -XPOST https://api.github.com/repos/${USER}/iter8/pulls -s -d \"{\\\"head\\\":\\\"iter8_exp\\\", \\\"base\\\":\\\"${BRANCH}\\\", \\\"body\\\":\\\"update baseline\\\", \\\"title\\\":\\\"Iter8 GitOps\\\"}\";\
-                  fi;\
-                  kubectl -n {{ .namespace }} apply -f istio-vs.yaml;\
-                 "
-                ]
+          url: https://api.github.com/repos/MY_ORG/iter8/actions/workflows/gitops-finish.yaml/dispatches
+          authType: Bearer
+          secret: iter8-token
+          body:  |
+            {
+              "ref":"master", 
+              "inputs":{
+                "filepath": "{{.filepath}}"
+              }
+            }
+          headers:
+          - name: Accept
+            value: application/vnd.github.v3+json
+
+      - task: common/bash
+        with:
+          script: |
+            kubectl -n {{ .namespace }} apply -f https://raw.githubusercontent.com/iter8-tools/iter8/master/samples/istio/gitops/istio-vs.yaml
```

Comment:
we don't want to hard code this URL

Reply:
I've replaced `iter8-tools` with `MY_ORG` so that it is replaced as part of the tutorial. There is no easy way to do otherwise without a script. Either script so can do git clone and local reference OR add to GH workflow but then need kubernetes credentials. This needs to be addressed in some way first.